core/commit.go

```diff
@@ -116,6 +116,10 @@ func makeCommitmentCollector(countCommitment commitmentCounter, retireSeq reques
 		}
 
 		pendingReq.Remove(request.Msg.ClientId)
+
+		// FIXME: Only stop the request timer if the request
+		// identifier it was started for matches the one of
+		// the request accepted for execution.
```

Comment:
I agree with this, thank you for spotting.

Reply:
Is there a possibility that a timer is left without stopping forever? I am not sure how a timer is stopped certainly at some point if `stopReqTimer()` doesn't stop the timer at this point.

Reply:
Is there a possibility that a timer is left without stopping forever? I am not sure how a timer is stopped certainly at some point if `stopReqTimer()` doesn't stop the timer at this point.

Reply:
Since correct clients only send their requests sequentially, a replica need to maintain one prepare and one request timer per client. If a newer request is received, the replica can assume the former request was completed and the client timers should be reused. In other words, replicas only maintain timers for the most recent request of each client.